Server/src/routes/vehicle.route.ts

```diff
@@ -368,7 +375,23 @@ export class VehicleRoute {
 				nearbyVehicles.map(async v => {
 					const pos = await VehicleService.getVehiclePosition(v)
 					const trackers = await database.trackers.getByVehicleId(v.uid)
-					const nearbySimplifiedVehicleHeading: number = await VehicleService.getVehicleTrackHeading(v, track)
+					const nearbyVehicleTrackKm: number | null = await VehicleService.getVehicleTrackDistanceKm(v)
+					if (!nearbyVehicleTrackKm) {
+						logger.error(`Could not compute track kilometer for vehicle with id ${v.uid}
+						 at track wit id ${v.trackId}`)
+						return {
+							id: v.uid,
+							name: v.name,
+							track: v.trackId,
+							type: v.typeId,
+							trackerIds: trackers.map(t => t.uid),
+							pos: pos ? { lat: GeoJSONUtils.getLatitude(pos), lng: GeoJSONUtils.getLongitude(pos) } : undefined,
+							percentagePosition: (await VehicleService.getVehicleTrackDistancePercentage(v)) ?? -1,
+							heading: await VehicleService.getVehicleHeading(v),
+							headingTowardsUser: undefined
+						}
```

Comment:
I am not quite sure, if that is the thing that should be returned at this point, but this had to be adjusted due to the major change of `getVehicleTrackHeading`. I did not want to mess up your code completely, so feel free to optimize here (maybe via a for-loop?).